inc/zoo/swar/SWAR.h

```diff
@@ -430,13 +430,49 @@ booleans(SWAR<NB, T> arg) noexcept {
 template<int NBits, typename T>
 constexpr auto
 differents(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
-    return booleans(a1 ^ a2);
+    return ~equals(a1, a2);
 }
 
+/*
+ * Returns a BooleanSWAR that contains a true value in each lane where the two
+ * inputs SWARs match.
+ */ 
 template<int NBits, typename T>
 constexpr auto
 equals(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
-    return ~differents(a1, a2);
+    using S = swar::SWAR<NBits, T>;
+    const auto h = S::MostSignificantBit, l = S::LeastSignificantBit;
+    const auto x = (a1 ^ a2).value();
+    return BooleanSWAR<NBits, T>(h & ~(x| ((x|h) -1)));
+}
+
+/*
+ * Returns a raw test value. If there are no null lanes present, will be all
+ * zeros.  If there is a null lane present, that lane will be all 1s, and any
+ * less significant lane will be all zeros. Anything more significant than the
+ * least significant null lane will be undefined.
+ * Less operations than equals()
+ */
+template<int NBits, typename T>
+constexpr auto
+leastSignificantZeroLane(SWAR<NBits, T> x) {
+    using S = swar::SWAR<NBits, T>;
+    const auto h = S::MostSignificantBit, l = S::LeastSignificantBit;
+    //const T x = a1.value();
+    return h & ( x.value() - l) & ~x.value();
+}
+
+/*
+ * As leastSignificantZeroLane, except we test against a given swar.
+ * Less operations than equals()
+ */ 
+template<int NBits, typename T>
+constexpr auto
+leastSignificantMatchingLane(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
+    using S = swar::SWAR<NBits, T>;
+    const auto h = S::MostSignificantBit, l = S::LeastSignificantBit;
+    const T x = a1 ^ a2;
+    return h & ( x - l) & ~x;
 }
```

Comment:
I am seeing that these functions ought not to be promoted right away to the main library `SWAR.h`, but as something more experiemental:
I am not sure the "raw test value" they return is useful by itself.
About `leastSignificantZeroLane`, this returns a `BooleanSWAR` since there is a conjunction with the most significant bit, so, the comment that describes the return value is wrong (the lane won't be all ones).  I do not like the nomenclature of "least significant blabla", but "**first**", this is how it is *used* in pokerbotic, Robin Hood, and the string operations.  I have no qualm about using a nomenclature that assumes or implies little endian byte encoding.  Yes, there are operations that use "least significant index" or the like, but in general, precisely because of the actual *use* it emerged that their right names are "first".
Do you agree on this nomenclature?

Reply:
Nomenclature changes: sure I can do that.
`pseudoBooleanSWAR` concept or similar covers these return values.  We just need to tighten up language around them and define it somewhere.
I think post discussion yesterday and TAOCP reread you consider these worth putting in the main library now?

Reply:
Not the main library.
I am beginning to have anxiety about the state of `SWAR.h`.  It is an error, long term, having implemented so many operations as members of `SWAR`, either class-members or instance members.
For example, we can't have `constexpr` constants of "intrinsic types", the SIMD types that compiler makers give you via autovectorization capabilities (GCC, Clang) or the classical architecure-specific headers.
If all implementations would be free standing functions, we could specialize them or customize them.
I prefer that we don't add to `SWAR.h` things that are not mature, for what passes as mature in this codebase, this will help welcome new users.
I have gained strong conviction that your original idea is necessary, that we totally need to express these techniques and to flesh out the potential new concept of PseudoBooleanSWAR or SIMD, just not in the main header.
Reasonable?

Reply:
I'm happy to break things apart, and generally prefer smaller 'chunks' of functionality grouped together.  
I'd also prefer to break that apart in some future PR because I find I often get sidequested on trying to figure out exactly what the right breakage boundary is &etc.